.pipelines/npm/npm-conformance-tests.yaml

```diff
@@ -1,270 +1,449 @@
+            then
+              az extension add --name aks-preview
+              az extension update --name aks-preview
+
+              # Enable Microsoft.ContainerService/AKSWindows2022Preview
+              az feature register --namespace Microsoft.ContainerService --name AKSWindows2022Preview
+              az provider register -n Microsoft.ContainerService
+
+              echo "creating WS22 Cluster";
+              az aks create \
+                  --resource-group $(RESOURCE_GROUP) \
+                  --name $(AZURE_CLUSTER) \
+                  --generate-ssh-keys \
+                  --windows-admin-username e2eadmin \
+                  --windows-admin-password alpha@numeric!password2 \
+                  --network-plugin azure \
+                  --vm-set-type VirtualMachineScaleSets \
+                  --kubernetes-version v1.23 \
+                  --node-vm-size Standard_D4s_v3 \
+                  --node-count 2
+
+              # don't schedule anything on the linux system pool
+              echo "Updating $(AZURE_CLUSTER) to not schedule anything on linux pool..."
+              az aks nodepool update \
+                --cluster-name $(AZURE_CLUSTER) \
+                -g $(RESOURCE_GROUP) \
+                -n nodepool1 \
+                --node-taints CriticalAddonsOnly=true:NoSchedule
+
+              echo "Adding Windows nodepool to $(AZURE_CLUSTER) to group $(RESOURCE_GROUP)"
+              az aks nodepool add \
+                  --resource-group $(RESOURCE_GROUP) \
+                  --cluster-name $(AZURE_CLUSTER) \
+                  --name awin22 \
+                  --os-type Windows \
+                  --os-sku Windows2022 \
+                  --node-vm-size Standard_D4s_v3 \
+                  --node-count 2
+            else
+              echo "Creating Linux Cluster"; 
+              az aks create --no-ssh-key \
+              --resource-group $(RESOURCE_GROUP) \
+              --name $(AZURE_CLUSTER) \
+              --network-plugin azure
+            fi
+
+            echo Cluster $(AZURE_CLUSTER)
+            echo Resource $(RESOURCE_GROUP)
+
+            az aks get-credentials -n $(AZURE_CLUSTER) -g $(RESOURCE_GROUP) --file ./kubeconfig
+            # get kubectl 
+            curl -LO https://dl.k8s.io/release/v1.20.0/bin/linux/amd64/kubectl
+            chmod +x kubectl
+
+            # deploy azure-npm
+            ./kubectl --kubeconfig=./kubeconfig apply -f https://raw.githubusercontent.com/Azure/azure-container-networking/master/npm/azure-npm.yaml
+
+            # swap azure-npm image with one built during run
+            ./kubectl --kubeconfig=./kubeconfig set image daemonset/azure-npm -n kube-system azure-npm=$IMAGE_REGISTRY/azure-npm:linux-amd64-$(TAG)
+
+            # swap NPM profile with one specified as parameter
+            ./kubectl --kubeconfig=./kubeconfig apply -f https://raw.githubusercontent.com/Azure/azure-container-networking/master/npm/profiles/$(PROFILE).yaml
```

Comment:
for ws22, having PROFILE as "v2-default-ws22" has unintended effect of failing to apply a configmap here